exercise/submission_models.py

```diff
@@ -290,6 +303,13 @@ def set_error(self):
     def is_graded(self):
         return self.status in (self.STATUS.READY, self.STATUS.UNOFFICIAL)
 
+    @property
+    def lang(self):
+        try:
+            return self.meta_data.get('lang', None)
+        except AttributeError:
```

Comment:
What would raise `AttributeError`?

Reply:
.get if meta_data == None for submissions created before these changes

Reply:
I see. Was commented here https://github.com/apluslms/a-plus/pull/588#discussion_r446917317  :)

Reply:
This is pretty good too. Maybe a comment saying "to handle cases where database includes null or non dictionary json"

```python
>>> json.loads('{}').get('foo', '<not found>')
'<not found>'
>>> json.loads('true').get('foo', '<not found>')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'bool' object has no attribute 'get'
```

Alternative would be checking that the `meta_data` is actually a dict, [like here](https://github.com/apluslms/a-plus/blob/master/exercise/templatetags/exercise.py#L241).